packages/react/src/file-upload/FileUpload.tsx

```diff
@@ -197,129 +239,185 @@ export const FileUpload = forwardRef<HTMLInputElement, FileUploadProps>(
 		} = getRootProps();
 
 		const {
+			/**
+			 * Warning! This ref is required for the button click to work in Safari and Firefox.
+			 * Also, linting doesn't complain about this TS error, but VS Code does so can't @ts-ignore etc.
+			 */
 			ref: dropzoneInputRef,
 			// We don't want the input to be `display: none`, as we are using visuallyHiddenStyles instead.
 			style: _unusedStyleProps,
 			...dropzoneInputProps
 		} = getInputProps();
 
-		const showFileLists = Boolean(
-			value.length || fileRejections.length || existingFiles?.length
-		);
-
 		const fileSummaryText = getFileListSummaryText([
 			...value,
-			...(existingFiles || []),
+			...existingFiles,
 		]);
 
+		const showFileLists = Boolean(value.length || existingFiles?.length);
+		const pluralAllRejections = allRejections.length > 1;
+
 		return (
 			<Field
 				label={label}
 				hideOptionalLabel={hideOptionalLabel}
 				required={Boolean(required)}
 				hint={hint}
-				message={message || errorSummary}
-				invalid={invalid || !!errorSummary}
+				message={message}
+				invalid={invalid}
 				id={id}
 			>
 				{(a11yProps) => {
 					return (
-						<Stack gap={1.5} {...dropzoneProps}>
-							<Flex
-								gap={1}
-								padding={1.5}
-								alignItems="center"
-								flexDirection="column"
-								border
-								rounded
-								css={styles}
-							>
-								<UploadIcon size="lg" color="muted" />
-								<input
-									{...dropzoneInputProps}
-									{...a11yProps}
-									{...consumerProps}
-									/**
-									 * Dropzone needs to set a ref to the input, but we _also_
-									 * need to forward a ref to the input so consumers can use it.
-									 * The mergeRef utility allows us to do this.
-									 */
-									ref={mergeRefs([forwardedRef, dropzoneInputRef])}
-									css={visuallyHiddenStyles}
-								/>
+						<>
+							<Box {...dropzoneProps}>
 								<Flex
-									flexDirection="column"
+									gap={1}
+									padding={1.5}
 									alignItems="center"
-									textAlign="center"
+									flexDirection="column"
+									border
+									rounded
+									css={styles}
 								>
-									<Text
-										color={isDragActive ? 'action' : 'muted'}
-										fontWeight="bold"
+									<UploadIcon size="lg" color="muted" />
+									<input
+										{...dropzoneInputProps}
+										{...a11yProps}
+										{...consumerProps}
+										/**
+										 * Dropzone needs to set a ref to the input, but we _also_
+										 * need to forward a ref to the input so consumers can use it.
+										 * The mergeRef utility allows us to do this.
+										 */
+										ref={mergeRefs([forwardedRef, dropzoneInputRef])}
+										css={visuallyHiddenStyles}
+									/>
+									<Flex
+										flexDirection="column"
```

Comment:
`Stack`?

Reply:
Yep, could be.

Reply:
Seeing these comments about using stack. I mainly wasn't using it since I was trying to make it work with the "alert" role div always being mounted which broke the spacing if I used stack for spacing out the elements.

Since the alert role has been replaced with the focus on mount functionality, the HTML can probably be simplified back to using stack for spacing now.

Reply:
Done